### DISCUSSION TITLE
5G sidelink simulation

### DISCUSSION THREAD

**Discussion by ana-luisa00** [eclipse/mosaic#297] [Q&A]
Hey everyone, I have been trying to use Eclipse MOSAIC with the LENA-5G module of ns3 but am unsure if this is even possible since I have had no luck in integrating it with Eclipse MOSAIC. 

Is it possible to use Eclipse MOSAIC to do any 5G sidelink simulation with NS3 or OMNET++ modules?

Thank you so much in advance

**Comment by xwh0625** [comment]
Hello，Is it possible to implement 5G features now?

**Reply by hoelger** [reply]
Currently there is the work to integrate ns3's LTE code. You can find the work-in-progress here:

https://github.com/eclipse-mosaic/mosaic/pull/501
https://github.com/mosaic-addons/ns3-federate/pull/11

This will pave the road to implement 5G features in the future :)

**Comment by hoelger** [comment]
The ns3 LTE feature is now merged into the main branches.

**Comment by hoelger** [comment]
Thanks to the hard work of @Jacenty00 we now have a first version where omnet is able to support 5G cellular communication!

See branches here:
https://github.com/eclipse-mosaic/mosaic/tree/1132-enable-lte-5g-in-omnet
https://github.com/mosaic-addons/omnetpp-federate/tree/1132-enable-lte-5g-in-omnet
